### PR TITLE
Fix bug with Events displaying

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Changes proposed in this pull request:
- Fix bug with `Events` displaying in `Service Catalog` and `Instances` views

**Related issue(s)**
See also https://github.com/kyma-project/kyma/issues/375

### CODE REVIEW

File: instances/src/commons/common.spec.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
+import { validateAsyncApiSpec } from './helpers';
+
+describe('validate asyncApiSpec', () => {
+  it('returns true', () => {
```
Comment (pkosiec) [inline instances/src/commons/common.spec.js]: More meaningful names

File: instances/src/components/ServiceInstanceDetails/ServiceInstanceTabs/ServiceInstanceTabs.component.js [hunk 2]
```diff
@@ -1,28 +1,28 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 
-import { Tabs, Tab } from '@kyma-project/react-components';
+import { Text, Tabs, Tab } from '@kyma-project/react-components';
 
 import Events from '../Events/Events.component';
-import ApiConsole from '../SwaggerApi/SwaggerApiConsole.component';
+import ApiReference from '../SwaggerApi/SwaggerApiReference.component';
```
Comment (pkosiec) [inline instances/src/components/ServiceInstanceDetails/ServiceInstanceTabs/ServiceInstanceTabs.component.js]: The ApiConsole should be used

File: catalog/src/components/ServiceClassDetails/ServiceClassDetails.component.js [hunk 3]
```diff
@@ -80,9 +80,13 @@ class ServiceClassDetails extends React.Component {
...
-                {serviceClass.serviceClass.content ||
-                serviceClass.serviceClass.apiSpec ||
-                serviceClass.serviceClass.asyncApiSpec ? (
+                {(serviceClass.serviceClass.content &&
+                  Object.keys(serviceClass.serviceClass.content).lenght) ||
+                (serviceClass.serviceClass.apiSpec &&
+                  Object.keys(serviceClass.serviceClass.apiSpec).lenght) ||
+                (serviceClass.serviceClass.asyncApiSpec &&
+                  Object.keys(serviceClass.serviceClass.asyncApiSpec)
+                    .lenght) ? (
```
Comment (pkosiec) [inline catalog/src/components/ServiceClassDetails/ServiceClassDetails.component.js]: length

File: instances/src/components/ServiceInstanceDetails/ServiceInstanceDetails.component.js [hunk 4]
```diff
@@ -52,12 +52,13 @@ class ServiceInstanceDetails extends React.Component {
...
             serviceInstanceRefetch={serviceInstance.refetch}
           />
-          {serviceClass ? (
-            serviceClass.content ||
-            serviceClass.apiSpec ||
-            serviceClass.asyncApiSpec ? (
-              <ServiceInstanceTabs serviceClass={serviceClass} />
-            ) : null
+          {serviceClass &&
+          ((serviceClass.content && Object.keys(serviceClass.content).lenght) ||
```
Comment (pkosiec) [inline instances/src/components/ServiceInstanceDetails/ServiceInstanceDetails.component.js]: length

File: instances/src/components/ServiceInstanceDetails/ServiceInstanceDetails.component.js [hunk 5]
```diff
@@ -52,12 +52,13 @@ class ServiceInstanceDetails extends React.Component {
...
+            (serviceClass.apiSpec &&
+              Object.keys(serviceClass.apiSpec).lenght) ||
```
Comment (pkosiec) [inline instances/src/components/ServiceInstanceDetails/ServiceInstanceDetails.component.js]: length

File: instances/src/components/ServiceInstanceDetails/ServiceInstanceDetails.component.js [hunk 6]
```diff
@@ -52,12 +52,13 @@ class ServiceInstanceDetails extends React.Component {
...
+            (serviceClass.asyncApiSpec &&
+              Object.keys(serviceClass.asyncApiSpec).lenght)) ? (
```
Comment (pkosiec) [inline instances/src/components/ServiceInstanceDetails/ServiceInstanceDetails.component.js]: length

File: catalog/src/components/ServiceClassDetails/Events/Events.component.js [hunk 7]
```diff
@@ -1,18 +1,44 @@
...
 
-import { NameSpan, VersionSpan, Code } from './styled';
+import { Bold, NameSpan, VersionSpan, Code } from './styled';
+
+import { validateAsyncApiSpec } from '../../../commons/helpers';
+
+function Events({ asyncApiSpec }) {
+  if (!validateAsyncApiSpec(asyncApiSpec)) {
+    return (
+      <Text color="#f44336" fontSize="14px">
```
Comment (michal-hudy) [inline catalog/src/components/ServiceClassDetails/Events/Events.component.js]: `Events` tab shouldn't be displayed in that case. User doesn't know what is `asyncApisSpec`.

File: catalog/src/components/ServiceClassDetails/ServiceClassTabs/ServiceClassTabs.component.js [hunk 8]
```diff
@@ -39,36 +37,45 @@ const ServiceClassTabs = ({ serviceClass }) => {
...
+                documentsByType[type].map((item, i) => {
+                  return item.source || item.Source ? (
+                    <div
+                      key={i}
+                      dangerouslySetInnerHTML={{
+                        __html: item.source || item.Source,
+                      }}
+                    />
+                  ) : (
+                    <Text key={i} color="#f44336" fontSize="14px">
```
Comment (michal-hudy) [inline catalog/src/components/ServiceClassDetails/ServiceClassTabs/ServiceClassTabs.component.js]: Same as for `asyncApiSpec`

File: instances/src/components/ServiceInstanceDetails/Events/Events.component.js [hunk 9]
```diff
@@ -1,17 +1,44 @@
...
 
-import { NameSpan, VersionSpan, Code } from './styled';
+import { Bold, NameSpan, VersionSpan, Code } from './styled';
+
+import { validateAsyncApiSpec } from '../../../commons/helpers';
+
+function Events({ asyncApiSpec }) {
+  if (!validateAsyncApiSpec(asyncApiSpec)) {
+    return (
+      <Text color="#f44336" fontSize="14px">
```
Comment (michal-hudy) [inline instances/src/components/ServiceInstanceDetails/Events/Events.component.js]: Same as for `ServiceClassDetails`

File: instances/src/components/ServiceInstanceDetails/ServiceInstanceTabs/ServiceInstanceTabs.component.js [hunk 10]
```diff
@@ -33,40 +33,52 @@ const ServiceInstanceTabs = ({ serviceClass }) => {
...
+                documentsByType[type].map((item, i) => {
+                  return item.source || item.Source ? (
+                    <div
+                      key={i}
+                      dangerouslySetInnerHTML={{
+                        __html: item.source || item.Source,
+                      }}
+                    />
+                  ) : (
+                    <Text key={i} color="#f44336" fontSize="14px">
```
Comment (michal-hudy) [inline instances/src/components/ServiceInstanceDetails/ServiceInstanceTabs/ServiceInstanceTabs.component.js]: Same as for `ServiceClassDetails`

File: catalog/src/commons/commons.spec.js [hunk 11]
```diff
@@ -38,3 +38,124 @@ describe('correctly random name', () => {
     expect(result).toEqual(true);
   });
 });
+
+describe('validate asyncApiSpec', () => {
+  it('returns true', () => {
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply catalog/src/commons/commons.spec.js]: 👌

File: catalog/src/commons/commons.spec.js [hunk 12]
```diff
@@ -18,23 +18,144 @@ const checkingPattern = names => {
 };
 
 describe('correctly adjectives', () => {
-  it('returns true if the all adjectives have a correctly pattern', () => {
+  it('should returns true if the all adjectives have a correctly pattern', () => {
```
Comment (pkosiec) [inline catalog/src/commons/commons.spec.js]: should return - without `s` suffix - please fix it in all descriptions

**Pull request merged**